### ISSUE TITLE
[React]: clickable Tile applies un-themed link that's inaccessible for g90, g100 themes

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon#3270]
## What package(s) are you using?

```
    "carbon-components": "^10.3.2",
    "carbon-components-react": "^7.3.2",
```

## Detailed description

Clickable `Tile` variation inserts a link that doesn't have the `bx--link` class and is therefore un-themed (and inaccessible for dark themes, as a result):

![Screen Shot 2019-07-02 at 4 31 47 PM](https://user-images.githubusercontent.com/9057921/60548103-e763cf80-9ce6-11e9-9847-38f71008e929.png)
